### PR TITLE
docs/consul: rename the Vault secret engine for Consul integration

### PR DESCRIPTION
**Pull request by krastin**
### Description

There was feedback from our Support team about customers facing an issue following the documentation from the [Vault for Consul: System integration](https://developer.hashicorp.com/consul/docs/k8s/deployment-configurations/vault/systems-integration) into the [Vault for Consul: Data integration](https://developer.hashicorp.com/consul/docs/k8s/deployment-configurations/vault/data-integration). The former defines a `consul`-named Vault secret engine, but the latter runs commands and config against a `secrets`-named Vault secret engine - no doubt a leftover from times where the default secret engine from running Vault in `-dev` mode was used to showcase functionality in the docs.

Furthermore, because of the secret engine name `consul`, the customers were confused and expected the secret engine to be of the _Consul_ type and not the _KV_ type.

With that in mind, this PR changes the docs in the following way:

- The Vault secret engine created for use with Consul has been renamed to `consul-kv`
    -  to stress the fact that it is a _KV_ type engine and not a _Consul_ type engine
    - to move away from the default secrets engine as this would be unadvisable to run in Production systems, and possibly even dangerous if the default secrets engine is indeed being used
- The path `secrets` is being used for all token-related configuration so that in Production environments, that specific path can be allowed/disallowed read/write access easier for separate admin and non-admin user roles.

### Testing & Reproduction steps

N/A

### Links

[Related Docs: Vault as the Secrets Backend (K8s)](https://developer.hashicorp.com/consul/docs/k8s/deployment-configurations/vault/)

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/k8s/deployment-configurations/vault/data-integration/bootstrap-token.mdx [hunk 1]
```diff
@@ -29,7 +29,7 @@ Prior to setting up the data integration between Vault and Consul on Kubernetes,
 First, generate and store the ACL bootstrap token in Vault. You will only need to perform this action once:
 
 ```shell-session
-$ vault kv put secret/consul/bootstrap-token token="$(uuidgen | tr '[:upper:]' '[:lower:]')"
+$ vault kv put consul-kv/secret/bootstrap-token token="$(uuidgen | tr '[:upper:]' '[:lower:]')"
```
[the remark this answers is not in the document]
Comment (krastin) [reply website/content/docs/k8s/deployment-configurations/vault/data-integration/bootstrap-token.mdx:32]: So CLI calls aim at `consul-kv/secret/XYZ-token` but API definitions and calls aim at `consul-kv/data/secret/XYZ-token`, it's a Vault specific thing where you have `[secret-engine]/data/[subpath]/`. Shall we put a note explaining this behaviour in each of the API calls; or just one note for when we create the secrets engine?

File: website/content/docs/k8s/deployment-configurations/vault/data-integration/index.mdx [hunk 2]
```diff
@@ -95,7 +95,7 @@ For example, if your Consul on Kubernetes servers need access to [Consul Server
       <CodeBlockConfig filename="license-policy.hcl">
 
       ```HCL
-      path "secret/data/consul/license" {
+      path "consul-kv/data/secret/license" {
```
[the remark this answers is not in the document]
Comment (krastin) [reply website/content/docs/k8s/deployment-configurations/vault/data-integration/index.mdx]: No, that was a mistake on my side - thanks for catching this! 😄

File: website/content/docs/k8s/deployment-configurations/vault/data-integration/bootstrap-token.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (krastin) [reply website/content/docs/k8s/deployment-configurations/vault/data-integration/bootstrap-token.mdx:32]: Good call - I've added a note in the index page
https://github.com/hashicorp/consul/pull/20049/files#diff-813dd2e4cf9d8739d05df920ee5b610e838b95ca55fbce0eec2a9130ec797217

**Comment by krastin** [comment]
List of manual backport PRs related to this one:

https://github.com/hashicorp/consul/pull/20092
https://github.com/hashicorp/consul/pull/20093
https://github.com/hashicorp/consul/pull/20094
https://github.com/hashicorp/consul/pull/20095
https://github.com/hashicorp/consul/pull/20096

**Comment by dhiaayachi** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
@krastin, a backport is missing for this PR [20049] for versions [1.15,1.16,1.17] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Pull request merged**